testutil/testclient/testblock/client.go

```diff
@@ -32,6 +33,31 @@ func NewLocalnetClient(ctx context.Context, t *testing.T) client.BlockClient {
 	return bClient
 }
 
+// NewAnyTimesCommittedBlocksSequenceBlockClient creates a new mock BlockClient.
+// This mock BlockClient will expect any number of calls to CommittedBlocksSequence,
+// and when that call is made, it returns the given BlocksObservable.
+func NewAnyTimesCommittedBlocksSequenceBlockClient(
+	t *testing.T,
+	blocksObs observable.Observable[client.Block],
+) *mockclient.MockBlockClient {
+	t.Helper()
+
+	// Create a mock for the block client which expects the LatestBlock method to be called any number of times.
+	blockClientMock := NewAnyTimeLatestBlockBlockClient(t, nil, 0)
```

Comment:
Just TOL but not actionable: feels like there's an opportunity to start chaining these depending on how the test framework evolves.